queries/erlang/highlights.scm

```diff
@@ -26,79 +62,68 @@
 
 [
   "catch"
-	"try"
-	"throw"
+  "try"
 ] @exception
+((atom) @exception (#any-of? @exception "error" "throw" "exit"))
+
+((atom) @boolean (#any-of? @boolean "true" "false"))
+
+;; Macros
+(macro_call_expr
+  "?" @macro
+  name: (_) @macro
+) @constant.macro
+
+;; Preprocessor
+(pp_define) @define
+[(pp_include) (pp_include_lib)] @include
+(_preprocessor_directive) @preproc
+
 ;;; module define
 [
   "module"
   "export"
+  "export_type"
+  "include"
+  "include_lib"
+  "behaviour"
+  "behavior"
 ] @include
-;;; operator
-[
-  ":"
-  ":="
-  "?"
-  "!"
-  "-"
-  "+"
-  "="
-  "->"
-  "=>"
-	"|"
-	;;;TODO
-	"$"
- ] @operator
+
+;; Records
+(record_field_expr _ @structure) @structure
+(record_field_name _ @structure) @structure
+(record_name name: _ @structure) @structure
+(record_decl name: _ @structure)
+(record_field name: _ @structure ty: _ @type)
+
+;; Type alias
+(type_alias "type" @type.definition)
+(spec "spec" @type.definition)
 
 (comment) @comment
-(string) @string
-(variable) @variable
+[(string) (binary)] @string
 
-(module_name
+(module_attribute
   (atom) @namespace
 )
 ;;; expr_function_call
-(expr_function_call
-  name: (computed_function_name) @function.call 
-) 
-
-(expr_function_call
-  arguments: (atom) @variable
-)
+(call expr: [(atom) (remote) (var)] @function)
 
-;;; map
-(map 
- (map_entry [
-   (atom)
-   (variable)
- ] @variable)
+;;; Parenthesized expression: (SomeFunc)(), only highlight the parens
+(call
+  expr: (paren_expr "(" @function ")" @function)
 )
 
+;;; function
+(external_fun) @function
+(internal_fun fun: (atom) @function)
+(function_clause name: (atom) @function)
 
-(tuple (atom) @variable)
-(pat_tuple ( pattern (atom) @variable))
+;;; attributes
+(attr_name name: (atom) @keyword)
 
-(computed_function_name) @function
-;;; case
-(case_clause
-  pattern: (pattern
-    (atom) @variable
-  )
-)
-(case_clause
-  body: (atom) @variable
-)
-
-;;; function
-(qualified_function_name
-  module_name: (atom) @attribute
-  function_name: (atom) @function
-)
-;; function
-(function_clause
-  name: (atom) @function)
-;;;lambda
-(lambda_clause
-  arguments:
-    (pattern) @variable
-)
+(var) @variable
+(atom) @constant
+(integer) @number
+(float) @float
```

Comment:
Move these up to the top, so the more specific captures override them.

Reply:
That doesn't seem to be enough, I need to also change their priority. It looks to me a bit like the highlighting prioritises more specific matches.

Reply:
Then I'd recommend leaving out the `(atom) @constant` capture, or at least make it much more specific (if that is possible). Otherwise it's easier to leave the other queries as they are for now (but of course changing the capture name as discussed).